lib/checkstl.cpp

```diff
@@ -2009,6 +2009,18 @@ void CheckStl::string_c_str()
                        ((Token::Match(tok->previous(), "%var% + %var% . c_str|data ( )") && tok->previous()->variable() && tok->previous()->variable()->isStlStringType()) ||
                         (Token::Match(tok->tokAt(-5), "%var% . c_str|data ( ) + %var%") && tok->tokAt(-5)->variable() && tok->tokAt(-5)->variable()->isStlStringType()))) {
                 string_c_strConcat(tok);
+            } else if (printPerformance && Token::simpleMatch(tok, "<<") && tok->astOperand2() && Token::simpleMatch(tok->astOperand2()->astOperand1(), ". c_str ( )")) {
```

Comment:
do we know that the `<<` will handle std::string ?
i.e. how about:

    QStringList x;
    x << str.c_str();

Reply:
No, that's why I check `if (strm && strm->variable() && strm->variable()->isStlType())`.

Reply:
ok